src/libPMacc/include/traits/GetNComponents.hpp

```diff
@@ -35,7 +35,11 @@ namespace traits
  * \return \p ::value as public with number of components (uint32_t)
  */
 template<typename T_Type, bool T_IsFundamental = boost::is_fundamental<T_Type>::value>
-struct GetNComponents;
+struct GetNComponents
+{
+    PMACC_CASSERT_MSG_TYPE( __GetNComponents_is_not_defined_for_this_type, T_Type, false ); 
+    static constexpr uint32_t dim = 0;
```

Comment:
This trait was unspecified by default. Now it is always specialized but if it is used on a unspecified  type the assert is triggered. 
The variable should only show the signature of this trait and avoid error message in form of `class has no member called `value`)

btw: `dim` should be called `value`